### PR TITLE
Enable the "Enable notebooks" button

### PR DESCRIPTION
**Pull request by srinathnarayanan**
Earlier, the "Enable Notebooks" button in the command bar was enabled only if the databaseAccount location was not a disallowed location. With these changes, the same check is made on both 
- database account location
- first write region of the database account

### CODE REVIEW

File: src/Explorer/Explorer.tsx [hunk 1]
```diff
@@ -1754,9 +1755,12 @@ export default class Explorer {
         this.isNotebooksEnabledForAccount(true);
         return;
       }
-      const isAccountInAllowedLocation = !disallowedLocations.some(
-        (disallowedLocation) => disallowedLocation === databaseAccountLocation
-      );
+
+      // databaseAccountLocation or firstWriteLocation are not disallowed
+      const isAccountInAllowedLocation =
+        disallowedLocations.indexOf(databaseAccountLocation) === -1 ||
```
Comment (swviswan) [inline src/Explorer/Explorer.tsx:1761]: >databaseAccountLocation [](http://example.com/codeflow?start=36&length=23)

do we need this check?

didnt we see that the control plane was using only the first write loc?

Comment (srinathnarayanan) [reply src/Explorer/Explorer.tsx]: modified it.

File: src/Explorer/Explorer.tsx [hunk 2]
```diff
@@ -1754,9 +1754,9 @@ export default class Explorer {
         this.isNotebooksEnabledForAccount(true);
         return;
       }
-      const isAccountInAllowedLocation = !disallowedLocations.some(
-        (disallowedLocation) => disallowedLocation === databaseAccountLocation
-      );
+
+      // firstWriteLocation should not be disallowed
+      const isAccountInAllowedLocation = disallowedLocations.indexOf(firstWriteLocation) === -1;
       this.isNotebooksEnabledForAccount(isAccountInAllowedLocation);
```
Comment (swviswan) [inline src/Explorer/Explorer.tsx:1760]: >isNotebooksEnabledForAccount [](http://example.com/codeflow?start=11&length=28)

can you add telemetry for how many ppl click this button and enable it?

Comment (srinathnarayanan) [reply src/Explorer/Explorer.tsx:1760]: there is already telemetry available for "CreateNotebookWorkspace" action which is triggered while enabling notebooks.

File: src/Explorer/Explorer.tsx [hunk 3]
```diff
@@ -1729,7 +1729,7 @@ export default class Explorer {
     }
 
     const databaseAccount = this.databaseAccount();
-    const databaseAccountLocation = databaseAccount && databaseAccount.location.toLowerCase();
+    const firstWriteLocation = databaseAccount?.properties?.writeLocations[0].locationName.toLowerCase();
```
Comment (gitanuj) [inline src/Explorer/Explorer.tsx:1732]: Do we need to check if `writeLocations` is defined and contains at least 1 element?

**Review by gitanuj — APPROVED** [review]

File: src/Explorer/Explorer.tsx (hunk 3 above)
Comment (srinathnarayanan) [reply src/Explorer/Explorer.tsx:1732]: Fixing it in the next iterations

**Pull request merged**